easybuild/easyblocks/e/elsi.py

```diff
@@ -0,0 +1,151 @@
+            self.log.info("Enabling PEXSI solver.")
+            self.cfg.update('configopts', "-DENABLE_PEXSI=ON")
+            if pexsi:
+                self.log.info("Using external PEXSI.")
+                self.cfg.update('configopts', "-DUSE_EXTERNAL_PEXSI=ON")
+                external_libs.append('pexsi')
+            else:
+                self.log.info("No external PEXSI specified as dependency, building internal PEXSI.")
+
+        slepc = get_software_root('SLEPc')
+        if slepc:
+            if self.cfg['build_internal_pexsi']:
+                # ELSI's internal PEXSI also builds internal PT-SCOTCH and SuperLU_DIST
+                raise EasyBuildError("Cannot use internal PEXSI with external SLEPc, due to conflicting dependencies.")
+            self.enable_sips = True
+            self.log.info("Enabling SLEPc-SIPs solver.")
+            self.cfg.update('configopts', "-DENABLE_SIPS=ON")
+            external_libs.extend(['slepc', 'petsc', 'HYPRE', 'umfpack', 'klu', 'cholmod', 'btf', 'ccolamd', 'colamd',
+                                  'camd', 'amd', 'suitesparseconfig', 'metis', 'ptesmumps',
+                                  'ptscotchparmetis', 'ptscotch', 'ptscotcherr', 'esmumps', 'scotch', 'scotcherr',
+                                  'stdc++', 'dl'])
+            if get_software_root('imkl') or get_software_root('FFTW'):
+                external_libs.extend(re.findall(r'lib(.*?)\.a', os.environ['FFTW_STATIC_LIBS%s' % self.env_suff]))
+            else:
+                raise EasyBuildError("Could not find FFTW library or interface.")
+
+        if get_software_root('imkl') or get_software_root('SCALAPACK'):
+            external_libs.extend(re.findall(r'lib(.*?)\.a', os.environ['SCALAPACK%s_STATIC_LIBS' % self.env_suff]))
+        else:
+            raise EasyBuildError("Could not find SCALAPACK library or interface.")
+
+        external_libs.extend(re.findall(r'-l(.*?)\b', os.environ['LIBS']))
+
+        self.cfg.update('configopts', "-DLIBS='%s'" % ';'.join(external_libs))
+        self.cfg.update('configopts', "-DLIB_PATHS='%s'" % ';'.join(lib_paths))
+        self.cfg.update('configopts', "-DINC_PATHS='%s'" % ';'.join(inc_paths))
+
+        super(EB_ELSI, self).configure_step()
+
+    def sanity_check_step(self):
+        """Custom sanity check for ELSI."""
+
+        libs = ['elsi', 'fortjson', 'MatrixSwitch', 'NTPoly', 'OMM']
+        modules = [lib.lower() for lib in libs if lib != 'OMM']
+        modules.extend(['omm_ops', 'omm_params', 'omm_rand'])
+
+        if self.cfg['build_internal_pexsi']:
+            modules.append('elsi_pexsi')
+            libs.extend(['pexsi', 'ptscotch', 'ptscotcherr', 'ptscotchparmetis',
+                         'scotch', 'scotcherr', 'scotchmetis', 'superlu_dist'])
+
+        if self.enable_sips:
+            modules.append('elsi_sips')
+            libs.append('sips')
+
+        libext = '.so' if self.cfg['build_shared_libs'] else '.a'
```

Comment:
Use get_shared_lib_ext, from easybuild.tools.systemtools instead of hardcoding .so